### PR TITLE
Assignment2: BenFristadSchool & mmoran314

### PR DESCRIPTION
**Pull request by BenFristadSchool**

### CODE REVIEW

**Comment by arose-13** [comment]
I'm gonna review this.

File: Logger.Tests/LogFactoryTests.cs [hunk 1]
```diff
@@ -6,5 +6,30 @@ namespace Logger.Tests;
...
+        // Assert
+        Assert.IsNull(logger);
+    }
+
+    [TestMethod]
+    public void CreateLogger_ReturnsLogger()
+    {
+        // Arrange 
+        LogFactory factory = new LogFactory();
+        factory.ConfigureFileLogger = "output.txt";
```
Comment (arose-13) [inline Logger.Tests/LogFactoryTests.cs:27]: `ConfigureFileLogger` should be used by the `CreateLogger` method.

**Review by arose-13 — COMMENTED** [review]
- There is an existing `BaseLogger` class. It needs an **auto property** to hold a class name. This property should be set in the `LogFactory` using an **object initializer**. ✔
- Create a `FileLogger` that derives from `BaseLogger`. It should take in a path to a file to write the log message to. When its `Log` method is called, it should **append** messages on their own line in the file. The output should include all of the following:
  - The current date/time ✔
  - The name of the class that created the logger ✔
  - The log level ✔
  - The message ✔
  - The format may vary, but an example might look like this "10/7/2019 12:38:59 AM FileLoggerTests Warning: Test message"
- The `LogFactory` should be updated with a new method `ConfigureFileLogger`. This should take in a file path and store it in a **private member**. It should use this when instantiating a new `FileLogger` in its `CreateLogger` method. ❌
- If the file logger has not be configured in the `LogFactory`, its `CreateLogger` method should return `null`. ✔
- Inside of `BaseLoggerMixins` implement **extension methods** on `BaseLogger` for
  - `Error`, ✔
  - `Warning`, ✔
  - `Information`, and ✔
  - `Debug`. ✔
  Each of these methods should take in a `string` for the message, as well as a **parameter array** of arguments for the message. Each of these extension methods is expected to be a shortcut for calling the `BaseLogger.Log` method, by automatically supplying the appropriate `LogLevel`. These methods should throw an exception if the `BaseLogger` parameter is null. There are a couple example unit tests to get you started.
- Use the nameof() operator when identifying the class name to the logger ✔
- Ensure you turn on Warnings as Errors (TreatWarningsAsErrors) ❌
- Ensure that you turn on code analysis (EnableNETAnalyzers) ❌
- Ensure that you turn on CodeAnalysisTreatWarningsAsErrors ❌
- Set `LangVersion` and the `TargetFramework` to the latest released versions available (preview versions optional) ✔
- Turn on Nullability (`Nullable`) ❌
- **All of the above should be unit tested.**

Only a couple small things need to be changed. The `LogFactory` needs to use the `ConfigureFileLogger` function when creating a `FileLogger`, and the "Warnings as Errors" and code analysis things need to be applied to the test project as well.

Other than that, good job!

File: Logger/BaseLoggerMixins.cs [hunk 2]
```diff
@@ -1,6 +1,51 @@
...
 public static class BaseLoggerMixins
 {
-    
+    public static void Error(BaseLogger logger, string message, params object[] args)
+    {
+        if (logger is null)
+        {
+            throw new ArgumentNullException(nameof(logger));
+        }
+        else
```
Comment (JLee35) [inline Logger/BaseLoggerMixins.cs:13]: `else` is unnecessary here.

File: Logger/LogFactory.cs [hunk 3]
```diff
@@ -2,9 +2,23 @@
 
 public class LogFactory
 {
+    private string? _Path;
+
     public BaseLogger CreateLogger(string className)
     {
-        
-        return null;
+        if(_Path == null)
```
Comment (JLee35) [inline Logger/LogFactory.cs:9]: Be consistent with null checks (using `_Path == null` vs `logger is null`).

File: Logger/Logger.csproj [hunk 4]
```diff
@@ -1,8 +1,12 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
-  <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <LangVersion>10.0</LangVersion>
-  </PropertyGroup>
+	<PropertyGroup>
+		<LangVersion>11.0</LangVersion>
+		<TargetFramework>netstandard2.0</TargetFramework>
```
Comment (JLee35) [inline Logger/Logger.csproj:5]:
```suggestion
		<TargetFramework>net7.0</TargetFramework>
```

**Review by JLee35 — COMMENTED** [review]
Good job!

**Comment by JLee35** [comment]
Good job!

**Pull request closed without merging**